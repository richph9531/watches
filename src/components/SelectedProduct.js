import React from 'react';
import PropTypes from 'prop-types';

import classes from '../styles/SelectedProduct.module.css';

const SelectedProduct = props => {
  const selectedWatchStrap = () => {
    const products = props.data.colorOptions;
    const product = products.filter(row => row.id === props.currentlySelectedStrapId);
    const response = product.length === 0 ? 1 : product[0].id;
    const filteredArray = products.filter(row => row.id === response);
    return filteredArray[0];
  };

  const TimeInHoursAndMinutes = () => {
    const hours = new Date().getHours() > 9 ? new Date().getHours() : `0${new Date().getHours()}`;
    const minutes = new Date().getMinutes() > 9 ? new Date().getMinutes() : `0${new Date().getMinutes()}`;
    return `${hours}:${minutes}`;
  };

  return (
  <div className={classes.SelectedProduct}>
    <img
      className={classes.WatchImage}
      src={`${process.env.PUBLIC_URL}/media/${selectedWatchStrap().imageUrl}`}
      alt={selectedWatchStrap().styleName}
    />
    { props.currentlySelectedFeature === 'Time'
      ? <div>
      <p className={classes.Time}>{TimeInHoursAndMinutes()}</p>
    </div>
      : <div>
      <img
        className={classes.HeartRateImage}
        src={`${process.env.PUBLIC_URL}/media/HeartRate.gif`}
      />
      <p className={classes.HeartRateText}>78</p>
    </div>
    }
  </div>
  );
};

SelectedProduct.propTypes = {
  data: PropTypes.object,
  showTimeOnWatch: PropTypes.bool,
  currentlySelectedStrapId: PropTypes.number,
  currentlySelectedFeature: PropTypes.string,
};

export default SelectedProduct;
