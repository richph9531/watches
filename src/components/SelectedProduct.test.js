import React from 'react';

import { render } from '@testing-library/react';
import SelectedProduct from './SelectedProduct';

test('Component renders correctly with Time selected and a valid strap colour', () => {
  const data = {
    data: {
      colorOptions: [
        { id: 1, styleName: 'Black Strap', imageUrl: 'BlackStrap.png' },
        { id: 2, styleName: 'Red Strap', imageUrl: 'RedStrap.png' },
      ],
    },
    currentlySelectedStrapId: 1,
    currentlySelectedFeature: 'Time',
  };
  const renderedElement = render(<SelectedProduct data={data.data}/>);
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Component renders correctly with Heart Rate selected and a valid strap colour', () => {
  const data = {
    data: {
      colorOptions: [
        { id: 1, styleName: 'Black Strap', imageUrl: 'BlackStrap.png' },
        { id: 2, styleName: 'Red Strap', imageUrl: 'RedStrap.png' },
      ],
    },
    currentlySelectedStrapId: 2,
    currentlySelectedFeature: 'HeartRate',
  };
  const renderedElement = render(<SelectedProduct data={data.data}/>);
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Component renders correctly with Heart Rate selected and an invalid strap colour', () => {
  const data = {
    data: {
      colorOptions: [
        { id: 1, styleName: 'Black Strap', imageUrl: 'BlackStrap.png' },
        { id: 2, styleName: 'Red Strap', imageUrl: 'RedStrap.png' },
      ],
    },
    currentlySelectedStrapId: 3,
    currentlySelectedFeature: 'HeartRate',
  };
  const renderedElement = render(<SelectedProduct data={data.data}/>);
  expect(renderedElement.baseElement).toMatchSnapshot();
});
