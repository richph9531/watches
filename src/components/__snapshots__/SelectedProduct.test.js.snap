// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly with Heart Rate selected and a valid strap colour 1`] = `
<body>
  <div>
    <div
      class="SelectedProduct"
    >
      <img
        alt="Black Strap"
        class="WatchImage"
        src="/media/BlackStrap.png"
      />
      <div>
        <img
          class="HeartRateImage"
          src="/media/HeartRate.gif"
        />
        <p
          class="HeartRateText"
        >
          78
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Component renders correctly with Heart Rate selected and an invalid strap colour 1`] = `
<body>
  <div>
    <div
      class="SelectedProduct"
    >
      <img
        alt="Black Strap"
        class="WatchImage"
        src="/media/BlackStrap.png"
      />
      <div>
        <img
          class="HeartRateImage"
          src="/media/HeartRate.gif"
        />
        <p
          class="HeartRateText"
        >
          78
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Component renders correctly with Time selected and a valid strap colour 1`] = `
<body>
  <div>
    <div
      class="SelectedProduct"
    >
      <img
        alt="Black Strap"
        class="WatchImage"
        src="/media/BlackStrap.png"
      />
      <div>
        <img
          class="HeartRateImage"
          src="/media/HeartRate.gif"
        />
        <p
          class="HeartRateText"
        >
          78
        </p>
      </div>
    </div>
  </div>
</body>
`;
